### PR TITLE
Update EOS examples for the new api on the Jungle 2.0 testnet

### PR DESCRIPTION
**Pull request by riccardopersiani**
### What this PR is:

Update the Oraclize **EOS examples** for being compliant with CDT on Jungle 2 and available for demanding people.

***

### To Do: 

__❍__  @gskapka  Please, could you review the README.md in every example folder?

__❍__  @gitmp01 Please, could you audit the code and check if the submodules has been done correctly?

__❍__ @bertani Please, could you audit the code and check especially the example `checkqueryid`, which has been modified with the new functions for emplacing and matching queryIds.

### CODE REVIEW

**Comment by bertani** [comment]
LGTM

**Comment by gskapka** [comment]
#### __Change 1:__

Before I check the individual `README`'s - can we first change the folder naming of everything, with words separated by hyphens everywhere, (like the main folder has: `eos-examples`). And also can the naming be more explicit?

Ie., what is `bitcoincomp`? Bitcoin composing? Bitcoin comparing? Bitco incompetance? `bitcoin-computation` is much more explicit.. Since the users of these examples may be new to the ecosystem, lets make things as easy as possible for them.

#### __Change 2:__

Can we also get a `README.md` in the main `eos-examples` dir. You can use [this one](https://github.com/oraclize/ethereum-examples) as a template.

#### __Change 3:__

Looking through the dirs, it's quite confusing since the individual examples `README`'s are all titled _"Testing..."_ and yet there're no tests in sight. Can the title of the `README` instead be something along the lines of _"Oraclize's Bitcoin Computation Example"_ or similar?

#### __Change 4:__

Can the links at the bottom of each `README` be formatted better? Using for e.g., the correct URL formatting of `[<text-here>](<url-here>)`. Then after each link can a sentence appear explaining where the link points to and why that might be useful to the user?

E.g. this:
  * :mag_right: :ledger: [**Blocks.io**:](https://jungle.bloks.io/) _Description of link goes here_

Instead of this:
  * :mag_right::ledger: **Blocks.io**: 

      https://jungle.bloks.io/

#### __Change 5:__

Due to __change 3__, the two emoji's used in the title should probably be changed. :wrench: and :construction: seem fitting for a _test_ repository, but less so for an _example_ repository (especially :construction: which may mislead users into thinking the example is under construction/not finished yet or something?)

#### __Change 6:__

Are there any instructions whatsoever for what a user needs to do with the examples given? For e.g., [here](github.com/oraclize/ethereum-examples) the main `README` gives a very brief description of what a user might do with the examples. Currently this repo has nothing of the sort.

***

Once the above changes have been made/address/discussed I'll go through and check each individual `README` itself.

**Comment by riccardopersiani** [comment]
**Change 1:**

Folder renamed (NB: folder naming is correct but changing the .cpp filename generates an empty .abi file).

**Change 2:**

The README.md is already present in the master folder.

**Change 3:**

Titles changed.

**Change 4:**

Links changed.

**Change 5:**

Icons removed.

**Change 6:**

Cannot see the link, but I think it will be useful for the users. Still thinking about it

File: eos-usd-price/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Oraclize's EOS/USD Price Example
+
+This repo is to demonstrate how you would work with the Oraclize **URL** datasource.
+
+The `eosusdprice` example allows you to *retrive the eos token price in usd*.
+
+## Notes
+
+Oraclize replies to your `oraclize_query` with the *price result* calling your `callback(...)`.
```
Comment (gskapka) [inline eos-usd-price/README.md]:
```suggestion
Oraclize replies to your `oraclize_query` by calling your `callback(...)` with the *price result*.
```

File: eos-usd-price/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+You can search your transaction ID in one of the following links to verify it:
+
+* :mag_right::ledger: [Blocks.io](https://jungle.bloks.io/): a block explorer for the Jungle 2.0 testnet.
```
Comment (gskapka) [inline eos-usd-price/README.md]:
```suggestion
* :mag_right::ledger: [Blocks.io](https://jungle.bloks.io/): A block explorer for the Jungle 2.0 testnet.
```

File: eos-usd-price/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+* :palm_tree::lion::palm_tree: [Jungle 2.0](https://monitor.jungletestnet.io/#home): a transaction explorer is available selecting *Get TX* on the Jungle 2.0 testnet website.
```
Comment (gskapka) [inline eos-usd-price/README.md]:
```suggestion
* :palm_tree::lion::palm_tree: [Jungle 2.0](https://monitor.jungletestnet.io/#home): A transaction explorer is available by selecting *Get TX* on the Jungle 2.0 testnet website.
```

File: random-datasource/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Oraclize's Random Datasource Example 
+
+This repo is to demonstrate how you would work with the Oraclize **random** datasource.
+
+The `randomsample` example allows you to *retrive a random number*.
```
Comment (gskapka) [inline random-datasource/README.md]:
```suggestion
The `randomsample.cpp` example allows you to *retrieve a random number*.
```

File: random-datasource/README.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+## Notes
+
+Oraclize replies to your `oraclize_newRandomDSQuery` with the *random number result* calling your `callback(...)`.
```
Comment (gskapka) [inline random-datasource/README.md]:
```suggestion
Oraclize replies to your `oraclize_newRandomDSQuery` by calling your `callback(...)` with the *random number result*.
```

File: random-datasource/README.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+You can search your transaction ID in one of the following links to verify it:
+
+* :mag_right::ledger: [Blocks.io](https://jungle.bloks.io/): a block explorer for the Jungle 2.0 testnet.
```
Comment (gskapka) [inline random-datasource/README.md]:
```suggestion
* :mag_right::ledger: [Blocks.io](https://jungle.bloks.io/): A block explorer for the Jungle 2.0 testnet.
```

File: random-datasource/README.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+* :palm_tree::lion::palm_tree: [Jungle 2.0](https://monitor.jungletestnet.io/#home): a transaction explorer is available selecting *Get TX* on the Jungle 2.0 testnet website.
```
Comment (gskapka) [inline random-datasource/README.md]:
```suggestion
* :palm_tree::lion::palm_tree: [Jungle 2.0](https://monitor.jungletestnet.io/#home): A transaction explorer is available by selecting *Get TX* on the Jungle 2.0 testnet website.
```

File: url-requests-computation/README.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# Oraclize's URL Requests Computation Example 
+
+This repo is to demonstrate how you would work with the Oraclize **computation** datasource.
+
+The `urlrequests` example allows you to *retrive the response of several URL requests*.
```
Comment (gskapka) [inline url-requests-computation/README.md]:
```suggestion
The `urlrequests.cpp` example allows you to retrieve the response of several URL requests.
```

File: url-requests-computation/README.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+## Notes
+
+Oraclize replies to your `oraclize_query` with the *respose body result* calling your `callback(...)`.
```
Comment (gskapka) [inline url-requests-computation/README.md]:
```suggestion
Oraclize replies to your `oraclize_query` by calling your `callback(...)` with the *respose body result*. 
```

File: url-requests-computation/README.md [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+This repo is to demonstrate how you would work with the Oraclize **computation** datasource.
+
+The `urlrequests` example allows you to *retrive the response of several URL requests*.
+
+## Notes
+
+Oraclize replies to your `oraclize_query` with the *respose body result* calling your `callback(...)`. 
+You can search your transaction ID in one of the following links to verify it:
+
+* :mag_right::ledger: [Blocks.io](https://jungle.bloks.io/): a block explorer for the Jungle 2.0 testnet.
```
Comment (gskapka) [inline url-requests-computation/README.md]:
```suggestion
* :mag_right::ledger: [Blocks.io](https://jungle.bloks.io/): A block explorer for the Jungle 2.0 testnet.
```

File: url-requests-computation/README.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
+# Oraclize's URL Requests Computation Example 
+
+This repo is to demonstrate how you would work with the Oraclize **computation** datasource.
+
+The `urlrequests` example allows you to *retrive the response of several URL requests*.
+
+## Notes
+
+Oraclize replies to your `oraclize_query` with the *respose body result* calling your `callback(...)`. 
+You can search your transaction ID in one of the following links to verify it:
```
Comment (gskapka) [inline url-requests-computation/README.md]:
```suggestion
You can search for your transaction ID in one of the following links to verify it:
```

File: url-requests-computation/README.md [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+
+* :mag_right::ledger: [Blocks.io](https://jungle.bloks.io/): a block explorer for the Jungle 2.0 testnet.
+
+* :palm_tree::lion::palm_tree: [Jungle 2.0](https://monitor.jungletestnet.io/#home): a transaction explorer is available selecting *Get TX* on the Jungle 2.0 testnet website.
```
Comment (gskapka) [inline url-requests-computation/README.md]:
```suggestion
* :palm_tree::lion::palm_tree: [Jungle 2.0](https://monitor.jungletestnet.io/#home): A transaction explorer is available by selecting *Get TX* on the Jungle 2.0 testnet website.
```

File: wolfram-alpha/README.md [hunk 13]
```diff
@@ -0,0 +1,15 @@
+# Oraclize's Wolfram Alpha Example
+
+This repo is to demonstrate how you would work with the Oraclize **WolframAlpha** datasource.
+
+The `wolframrand` example allows you to *retrive a random number from 1 to 6; and, if the number is
```
Comment (gskapka) [inline wolfram-alpha/README.md]:
```suggestion
The `wolframrand.cpp` example allows you to retrieve a random number from 1 to 6; and, if the number is
```

File: wolfram-alpha/README.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+not 6, the callback launches a new query*.
+
+## Notes
+
+Oraclize replies to your `oraclize_query` with the *random number between 1 and 6 result* calling your `callback(...)`.
```
Comment (gskapka) [inline wolfram-alpha/README.md]:
```suggestion
Oraclize replies to your `oraclize_query` by calling your `callback(...)` with the *random number between 1 and 6 result*.
```

File: wolfram-alpha/README.md [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+You can search your transaction ID in one of the following links to verify it:
```
Comment (gskapka) [inline wolfram-alpha/README.md]:
```suggestion
You can search for your transaction ID in one of the following links to verify it:
```

File: wolfram-alpha/README.md [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+
+* :mag_right::ledger: [Blocks.io](https://jungle.bloks.io/): a block explorer for the Jungle 2.0 testnet.
```
Comment (gskapka) [inline wolfram-alpha/README.md]:
```suggestion
* :mag_right::ledger: [Blocks.io](https://jungle.bloks.io/): A block explorer for the Jungle 2.0 testnet.
```

File: wolfram-alpha/README.md [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+
+* :palm_tree::lion::palm_tree: [Jungle 2.0](https://monitor.jungletestnet.io/#home): a transaction explorer is available selecting *Get TX* on the Jungle 2.0 testnet website.
```
Comment (gskapka) [inline wolfram-alpha/README.md]:
```suggestion
* :palm_tree::lion::palm_tree: [Jungle 2.0](https://monitor.jungletestnet.io/#home): A transaction explorer is available by selecting *Get TX* on the Jungle 2.0 testnet website.
```

File: wolfram-alpha/README.md [hunk 18]
```diff
@@ -0,0 +1,15 @@
+# Oraclize's Wolfram Alpha Example
+
+This repo is to demonstrate how you would work with the Oraclize **WolframAlpha** datasource.
+
+The `wolframrand` example allows you to *retrive a random number from 1 to 6; and, if the number is
+not 6, the callback launches a new query*.
```
Comment (gskapka) [inline wolfram-alpha/README.md]:
```suggestion
not 6, the `callback` launches a new query.
```

**Comment by riccardopersiani** [comment]
Instruction set added in every example

**Comment by riccardopersiani** [comment]
Node updated.
Every example tested successfully.

**Pull request merged**